src/VisualStudio/Core/Def/Implementation/UnusedReferences/Dialog/UnusedReferencesTableProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.UnusedReferences;
+using Microsoft.Internal.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.UnusedReferences.Dialog
+{
+    [Export(typeof(UnusedReferencesTableProvider))]
+    internal partial class UnusedReferencesTableProvider
+    {
+        private readonly ITableManager _tableManager;
+        private readonly IWpfTableControlProvider _tableControlProvider;
+        private readonly UnusedReferencesDataSource _dataSource;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public UnusedReferencesTableProvider(
+            ITableManagerProvider tableMangerProvider,
+            IWpfTableControlProvider tableControlProvider)
+        {
+            _tableManager = tableMangerProvider.GetTableManager(nameof(UnusedReferencesDataSource));
+            _tableControlProvider = tableControlProvider;
+
+            _dataSource = new UnusedReferencesDataSource();
+            _tableManager.AddSource(_dataSource, UnusedReferencesColumnDefinitions.ColumnNames);
```

Comment:
So we only create these once, so if the dialog is closed and opened again, we're reusing the prior data sources? Is that problematic?

Reply:
My worry here is it seems like multiple runs could accidentally polute information between each other if we're not careful, but maybe this is how this particular VS API works.

(the evil person in me is wondering what this would look like the if the dialog wasn't modal...)

Reply:
I got away with a lot because we were modal and only ever had one Sink at a time. Going to simplify and correct this bit.